auto_gptq/utils/marlin_utils.py

```diff
@@ -31,7 +33,7 @@ def prepare_model_for_marlin_load(
     device_map,
 ):
     # The model (e.g. model.safetensors) is already serialized in the Marlin format, load it directly.
-    if hasattr(quantize_config, "is_marlin_format") and quantize_config.is_marlin_format:
+    if hasattr(quantize_config, FORMAT_FIELD) and quantize_config.format == FORMAT.MARLIN:
```

Comment:
the quant_config passed here is actually BaseQuantizeConfig in all code paths. So we don't even need the hasattr() check. and compat is taken care of in BaseQuantizeConfig.from_pretrained(). 

I will refractor so BaseQuantizeConfig is moved from modeling/ to quantization/ to avoid circular imports and make the arg types explicitly clear.